doc/dev/BUILD.md

```diff
@@ -0,0 +1,127 @@
+---
+title: Building TileDB from source
+---
+
+## Prerequisites
+
+* CMake 3.21 or greater
+* C++ compiler with C++20 support
+    * The minimum compiler versions TileDB is being tested in CI are:
+        * MSVC from Visual Studio 2019 16.11
+        * GCC 10.3 on Linux and Windows via MinGW
+        * Apple Clang 14
+* Git (required by vcpkg)
+* curl (required by vcpkg on non-Windows)
+
+### Dependency management with vcpkg
+
+TileDB uses [vcpkg](https://vcpkg.io) to download and build its dependencies. By default, the vcpkg repository will be cloned automatically when configuring the build tree. To avoid repeatedly cloning the repository, and take full advantage of vcpkg's [binary caching](https://learn.microsoft.com/en-us/vcpkg/users/binarycaching) feature, you are recommended to clone the vcpkg repository manually and set the `VCPKG_ROOT` environment variable on your machine to the path of the repository.
+
+> [!NOTE]
+> If you have set `VCPKG_ROOT` on your machine and encountered errors like `error: no version database entry for <port> at <version>` it means that your vcpkg repository is out of date. You can update it by running `git pull` in the repository directory.
```

Comment:
This seems like a misfeature. If I'm using a lovely package manager, why doesn't it manage the package by downloading it with git? And why does it do so only when I set a magic environment variable and not otherwise?

Reply:
IIRC I added this note after hitting this error when working on #4553. The vcpkg `builtin-baseline` got updated to a recent commit, but that commit did not exist in my repository in `VCPKG_ROOT`.

> manage the package by downloading it with git

It's not the package, it's the database with the package versions. I guess vcpkg does not run `git fetch` by itself for performance reasons.